### ISSUE TITLE
New ignore behavior issue tracker

### ISSUE THREAD

**Issue by MrPetovan** [friendica/friendica#4960]
Ignoring a user should only:
- Hide their top-level posts from the `network/` page.

However, at the moment:
- [ ] Top level posts show in the `update_network/` process (reloading the `network/` page makes them disappear again).
- [ ] Posts don't show at all in `display/`, without displaying an error.

**Comment by tobiasd** [comment]
Their postings appear also in the result from the API

**Comment by MrPetovan** [comment]
Thanks for the reminder.

**Comment by annando** [comment]
Just to be sure: You are speaking of personal contacts (not the public ones) and the button to ignore this contact?

**Comment by tobiasd** [comment]
Yes, personal contacts currently ignored. The behavior had changed to a rather buggy state.

**Comment by annando** [comment]
What is the current state?

**Comment by tobiasd** [comment]
Something new for this, postings of ignored people now appear in both network stream, sorted by last comment and sorted by posting date.

So it is getting worse.

**Comment by annando** [comment]
How have you ignored them? Strange thing is: I haven't changed anything at the network page, only "display", "profile" and "notes".

**Comment by tobiasd** [comment]
Unter `/contacts` where the correct indicator "Currently Ignored" is still shown for the contact. And no, the posting was no posting to a forum.

**Comment by annando** [comment]
Okay. Possibly I have to add some new side table for blocking and ignoring contacts.This could also be the basis for the successor of superblock.

**Comment by MrPetovan** [comment]
I approve. There should be a global contact table, and a `user-contact` table with the relationships between  users and contacts, even if they aren't following them.

**Comment by MrPetovan** [comment]
Discussion here: https://forum.friendi.ca/display/ec054ce7105b7eb0eb2f21c692219031

**Comment by annando** [comment]
The new blocking feature is now in the core. Does anyone disagree to close this issue?

**Comment by tobiasd** [comment]
> The new blocking feature is now in the core. Does anyone disagree to
> close this issue?

Yes. First it should be tested if it is now working as expected ;-)

**Comment by MrPetovan** [comment]
What is the relationship between the blocking feature and the ignore feature shenanigans?

**Comment by tobiasd** [comment]
Is it expected that I see a comment to a thread by a ignored account when they are not the thread initiator?

**Comment by annando** [comment]
Yes, this is expected - and the difference between "block" and "ignore". "ignore" is for people who just post too much but you don't want to drop the connection. Ignored people are able to comment and to write private messages. Blocked persons will be blocked for every incoming stuff.

**Comment by MrPetovan** [comment]
Also posts by ignored contacts should still be received, just not displayed until the ignore status is lifted. Posts by blocked contacts are never linked with the user in the first place.

**Comment by annando** [comment]
We should exactly document the wanted behaviour, since currently the system doesn't behave exactly like this.

**Comment by tobiasd** [comment]
Yes, please update the existing documentation with the new behavior!

**Comment by MrPetovan** [comment]
Let's attempt at something. I believe the ignore feature can serve two purposes depending on the connection with the contact:
- No relationship: The goal is to shadow-block them, not seeing any content from them without alerting them to the fact they've been moderated. This is the equivalent of Twitter's Mute feature. 
- Existing relationship: The goal is to maintain the relationship without being exposed to their content unintentionally. Foreign language issues are probably better handled with the `langfilter` addon.

To this extent:
- Communication from/to ignore contacts is maintained.
- Top-level posts for ignored contacts shouldn't appear in feeds (network, community) at all, whether it is because they've been reshared or because one of your non-ignored contacts replied to one of their top-level posts.
- The content of their comments on displayed threads shouldn't appear by default. It may be an additional case for the content filter, or a new complete post filter à la Twitter to avoid seeing the avatar and display name. however this can pose problem for thread understanding.
- No notifications should be received from this contact.
- Posts by this account can still be accessed intentionally (`/display` module, contact's profile and network posts).
- The status should be seamlessly reversible, ignoring a contact should hide their content on the next refresh, unignoring a contact will restore normal display of their content on the next refresh as well. No content can be lost, although missed notifications won't be restored.

What do you think? Am I missing something?

**Comment by annando** [comment]
I'm unsure about the behaviour in comments. I would think that the difference between "block" and "ignore" would be that "block" completely blocks incoming content, while "ignore" mostly only had an impact on incoming top level posts.

**Comment by MrPetovan** [comment]
Why just top-level posts though? What is the rationale?

**Comment by AlfredSK** [comment]
Just my thoughts from a user's point of view. I expect 'ignore' to completely mute the user. Means, I don't see **any** content from the ignored user. But the user can receive my content. 
And I expect 'block' to **additionally** prevent sending posts and comments directly to the blocked user.

I don't expect any difference in the ignore-/block-behavior between users in my contacts and users not in my contacts.

**Comment by MrPetovan** [comment]
> I don't expect any difference in the ignore-/block-behavior between users in my contacts and users not in my contacts.

I don't either, I was just highlighting the different use cases, but the actual effects of the ignore feature I outlined aren't depending on the relationship.

> I don't see **any** content from the ignored user

Even if it breaks some threads? Would you still want to know a comment has been muted without showing the user? Would you be fine with seeing the user but not the comment content? I'm trying to get a sense of how much/little we should show from ignored users.

**Comment by AlfredSK** [comment]
Well, I can only speak for myself. Others might have different expectations.

> Even if it breaks some threads? 
Yes. 

> Would you still want to know a comment has been muted without showing the user?
No. I don't want to but I could get used to it when the comments are collapsed like the blockem add-on does it.

> Would you be fine with seeing the user but not the comment content?
Like above. I can imagine to implement the blockem behavior for comments. 

At the moment my workflow is like this...
1. content of user is temporarily annoying -> blockem add-on to collapse the posts and comments, but being able to read content on click
2. content of user is always annoying but user is not really a bad actor -> ignore user
3. Geez! I don't want to have any contact with that individual -> block user

**Comment by MrPetovan** [comment]
Thank you, I know you're only speaking for yourself but your opinion is still important to me.

Does the blockem addon also collapse the author avatar and display name? Or is it just the content itself like trigger warnings and language filter?

**Comment by AlfredSK** [comment]
Blockem has the same behavior like the language filter or the adv. content filter. You still see the avatar and the username but the content (body of item) is hidden.

**Comment by MrPetovan** [comment]
Ok thanks.

**Comment by annando** [comment]
Concerning the blocking: We cannot completely prevent that blocked users will get the content. When it was a public post and other persons on the same server has subscribed as well - or if the post was relayed, then the content will be available for blocked users as well.

So I would see the major difference between "block" and "ignore" in the receiver. Means: "ignore" only for toplevel posts, "block" for posts, comment and private messages.

I see "ignore" mostly for some: "I really like that person and I like their comments - but please prevent me from receiving all the cat content".

**Comment by MrPetovan** [comment]
The problem is that there is no in-between step in your concept of "block" and "ignore". There is no "I don't want to see their top-level posts and comments but I'm fine with them accessing my public or follower-only content. So I believe it removes some form of agency.
